bib/internal/uploader/aws.go

```diff
@@ -12,19 +12,51 @@ import (
 	"github.com/osbuild/images/pkg/cloud/awscloud"
 )
 
+func checkAWSCredentialFileExists() (bool, error) {
+	// SMELL make this path configurable?
+	credentialFile := "$HOME/.aws/credentials"
+
+	expandedFilename := os.ExpandEnv(credentialFile)
+	_, err := os.Stat(expandedFilename)
+	if err != nil {
+		return false, fmt.Errorf("AWS credential file not found: %s", expandedFilename)
```

Comment:
Super nitpick: When writing the tests for this I noticed that the error is a bit misleading, this can result in `AWS credential file not found: stat /tmp/TestCheckAWSCredentialFileExistsNotExisting694989933/001/.aws/credentials: permission denied` so maybe something more like `cannot use aws file` as the prefix?

Reply:
I would love if we would test as much as possible by default. This function is a nice one, I attached a test with an example of what I have in mind (it's a bit rough, a table driven test or multiple ones are better :)

[0001-uploader-create-simple-unit-test.patch.txt](https://github.com/osbuild/bootc-image-builder/files/13724250/0001-uploader-create-simple-unit-test.patch.txt)